crates/nargo/src/cli/mod.rs

```diff
@@ -143,7 +162,19 @@ fn write_inputs_to_file<P: AsRef<Path>>(
         dir_path
     };
 
-    let serialized_output = format.serialize(w_map)?;
+    // We must insert the return value into the `InputMap` in order for it to be written to file.
+    let serialized_output = match return_value {
+        // We don't want to modify the original map as this is an I/O implementation detail.
+        // We must then clone the map before insertion.
```

Comment:
I don't think I'd call it an I/O implementation detail when it impacts our public api for the verifier.toml file

Reply:
hmm, that's true. I was meaning relative to the core CLI proving/verifying logic but I see what you mean. Will update.